src/spec-common/injectHeadless.ts

```diff
@@ -503,18 +503,26 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 				// doesn't get interleaved with the output of other commands.
 				const printMode = name ? 'off' : 'continuous';
 				const env = { ...(await remoteEnv), ...(await secrets) };
-				const { cmdOutput } = await runRemoteCommand({ ...lifecycleHook, output: infoOutput }, containerProperties, typeof postCommand === 'string' ? ['/bin/sh', '-c', postCommand] : postCommand, remoteCwd, { remoteEnv: env, pty: true, print: printMode });
 
-				// 'name' is set when parallel execution syntax is used.
-				if (name) {
-					infoOutput.raw(`\x1b[1mRunning ${name} from ${userCommandOrigin}...\x1b[0m\r\n${cmdOutput}\r\n`);
-				}
+				try {
+					const { cmdOutput } = await runRemoteCommand({ ...lifecycleHook, output: infoOutput }, containerProperties, typeof postCommand === 'string' ? ['/bin/sh', '-c', postCommand] : postCommand, remoteCwd, { remoteEnv: env, pty: true, print: printMode });
 
-				infoOutput.event({
-					type: 'progress',
-					name: progressName,
-					status: 'succeeded',
-				});
+					// 'name' is set when parallel execution syntax is used.
+					if (name) {
+						infoOutput.raw(`\x1b[1mRunning ${name} of ${lifecycleHookName} from ${userCommandOrigin}...\x1b[0m\r\n${cmdOutput}\r\n`);
+					}
+				} catch (err) {
```

Comment:
so basically what i understood is that you'd like me to push this code of catching the error inside `runSingleCommand` method to be after https://github.com/devcontainers/cli/blob/main/src/spec-common/injectHeadless.ts#L531 i.e. await Promise.all(commands), in the catch block that's written after it

Reply:
I think we are very close in https://github.com/devcontainers/cli/pull/852. I suggest we continue there since that has all the history of the discussion. I have pushed the suggested fixes there, please take a look and leave a review. Thanks!

Reply:
closing this as #852 was already merged.